tests/flow_counter/flow_counter_utils.py

```diff
@@ -0,0 +1,278 @@
+    """
+    items = route_pattern.split('|')
+    if len(items) == 2:
+        dut.command('sudo config flowcnt-route pattern remove {} --vrf {}'.format(items[1], items[0]))
+    elif len(items) == 1:
+        dut.command('sudo config flowcnt-route pattern remove {}'.format(items[0]))
+    else:
+        logger.error('Invalid route pattern {}'.format(route_pattern))
+
+
+def clear_route_flow_counter(dut):
+    """Clear all route flow counter statistics
+
+    Args:
+        dut (object): DUT object
+    """
+    dut.command('sonic-clear flowcnt-route')
+
+
+def clear_route_flow_counter_by_pattern(dut, route_pattern):
+    """Clear route flow counter statistics by pattern
+
+    Args:
+        dut (object): DUT object
+        route_pattern (str): Route pattern. e.g. "1.1.1.0/24", "2000::/64", "Vrf1|2.2.2.0/24"
+    """
+    items = route_pattern.split('|')
+    if len(items) == 2:
+        dut.command('sonic-clear flowcnt-route pattern {} --vrf {}'.format(items[1], items[0]))
+    elif len(items) == 1:
+        dut.command('sonic-clear flowcnt-route pattern {}'.format(items[0]))
+    else:
+        logger.error('Invalid route pattern {}'.format(route_pattern))
+
+
+def clear_route_flow_counter_by_route(dut, prefix):
+    """Clear route flow counter statistics by route
+
+    Args:
+        dut (object): DUT object
+        prefix (str): Prefix pattern. e.g. "1.1.1.0/24", "2000::/64", "Vrf1|2.2.2.0/24"
+    """
+    items = prefix.split('|')
+    if len(items) == 2:
+        dut.command('sonic-clear flowcnt-route route {} --vrf {}'.format(items[1], items[0]))
+    elif len(items) == 1:
+        dut.command('sonic-clear flowcnt-route route {}'.format(items[0]))
+    else:
+        logger.error('Invalid prefix pattern {}'.format(prefix))
+
+
+def parse_route_flow_counter_stats(dut):
+    """Parse command output of "show flowcnt-route stats"
+
+    Args:
+        dut (object): DUT object
+
+    Returns:
+        dict: Parsed result. e.g. {'1.1.1.0/24': {'packets': '5', 'bytes': '4500'}}
+    """
+    stats_list = dut.show_and_parse('show flowcnt-route stats')
+    parse_result = {}
+    for stats in stats_list:
+        if stats['vrf'] == 'default':
+            key = stats['matched routes']
+        else:
+            key = '|'.join([stats['vrf'], stats['matched routes']])
+        parse_result[key] = {
+            'packets': stats['packets'],
+            'bytes': stats['bytes']
+        }
+    return parse_result
+
+
+def verify_route_flow_counter_stats(expect_stats, actual_stats):
+    """Verify actual statistic with expected statistic
+
+    Args:
+        expect_stats (dict): Expected stats. e.g. {'1.1.1.0/24': {'packets': '5', 'bytes': '4500'}}
+        actual_stats (dict): Actual stats. e.g. {'1.1.1.0/24': {'packets': '5', 'bytes': '4500'}}
+
+    Returns:
+        bool: Match if True.
+    """
+    logger.info('Expected stats: {}'.format(expect_stats))
```

Comment:
here I suggest to exit the function only once(at the end), so we can set the rc and message when needed and return it at the end

Reply:
Prefer to keep this. There is a "for loop in for loop" in this function, it is hard to use rc in the inner loop. 
```python
    for key, value in expect_stats.items():
        if key not in actual_stats:
            return False, 'Failed to find {} in result'.format(key)

        for stats_type, expect_value in value.items():
            if int(expect_value) != int(actual_stats[key][stats_type].replace(',', '')):
                return False, 'Expected {} value of {} is {}, but got {}'.format(stats_type, key, expect_value, actual_stats[key][stats_type])
```